src/components/edit/mail.edit.tsx

```diff
@@ -0,0 +1,44 @@
+import { useForm } from 'react-hook-form';
+import { useUserContext } from '../../api/contexts/user.context';
+import Form from '../../stories/form/Form';
+import StyledInput from '../../stories/form/StyledInput';
+import StyledButton, { StyledButtonColors } from '../../stories/StyledButton';
+import { Utils } from '../../utils';
+import Validations from '../../validations';
+
+interface MailEditProps {
+  onSubmit: () => void;
+}
+
+interface FormData {
+  email: string;
+}
+
+export function MailEdit({ onSubmit }: MailEditProps): JSX.Element {
+  const {
+    control,
+    handleSubmit,
+    formState: { isValid, errors },
+  } = useForm<FormData>();
+  const { changeMail } = useUserContext();
+
+  async function saveUser({ email }: FormData): Promise<void> {
+    changeMail(email).then(onSubmit);
+  }
+
+  const rules = Utils.createRules({
+    email: Validations.Required && Validations.Mail,
+  });
```

Comment:
If you copied the `createRules` from payment page, rule "concat" should be use like
```ts
  const rules: any = createRules({
    mail: [Validations.Required, Validations.Mail],
  });
```

Reply:
that is why it was not working as expected ;) thanks